src/auth/IdBroker.php

```diff
@@ -45,10 +47,28 @@ public function getAuthenticatedUser(string $username, string $password)
             'password' => $password,
         ]);
         $statusCode = $result['statusCode'] ?? null;
-        if (intval($statusCode) === 200) {
-            unset($result['statusCode']);
-            return $result;
+        switch (intval($statusCode)) {
+            
+            case 200: // Credentials were acceptable.
+                unset($result['statusCode']);
+                /*
+                 * Make sure all values are arrays themselves for simplesamlphp compatibility
+                 */
+                foreach ($result as $key => $value) {
+                    if ( ! is_array($value)) {
+                        $result[$key] = [$value];
+                    }
+                }
+                return $result;
+            
+            case 400: // Credentials were NOT acceptable.
+                return null;
+
+            default:
+                throw new \Exception(sprintf(
+                    'Unexpected response from ID Broker: %s',
+                    var_export($result->toArray(), true)
```

Comment:
Scrutinizer complained that `$result` would already be an array (and so calling `toArray()` wouldn't work here). I think we could instead have line 70 read `var_export((array)$result, true)` if we want to ensure `$result` is an array. Did you happen to find a way to trigger this part of the code to run?
(As a side note, I don't remember if I wrote this part or if you did, @fillup, so if this doesn't look familiar to you, I'll assume I did, and will go ahead and try to fix it.)